src/main/java/school/hei/haapi/service/ReceiptGenerationService.java

```
@@ -0,0 +1,178 @@
+                .startRequest(Instant.now())
+                .request(zipReceiptsRequest)
+                .build()));
+
+    return new ZipReceiptsStatistic()
+        .fileCountInZip(
+            paymentRepository.countByCreationDatetimeBetweenOrderByCreationDatetimeAsc(
+                zipReceiptsRequest.getFrom(), zipReceiptsRequest.getTo()));
+  }
+
+  private Context loadPaymentReceiptContext(Fee fee, Payment payment) {
+    Resource logo = classPathResourceResolver.apply("HEI_logo", ".png");
+    Context context = new Context();
+    List<Payment> paidPaymentsBefore =
+        paymentService.getByFeeIdOrderByCreationDatetimeAsc(fee.getId());
+    PaidFeeReceiptDataProvider dataProvider =
+        new PaidFeeReceiptDataProvider(fee.getStudent(), fee, payment, paidPaymentsBefore);
+
+    context.setVariable("logo", base64Converter.apply(logo));
+    context.setVariable("paymentAuthorName", dataProvider.getEntirePaymentAuthorName());
+    context.setVariable("receiptNumber", payment.getSequence().getStringSequence());
+    context.setVariable("totalAmount", numberToReadable(dataProvider.getFeeTotalAmount()));
+    context.setVariable("paymentDate", dataProvider.getPaymentDate());
+    context.setVariable("paymentAmount", numberToReadable(dataProvider.getTotalPaymentAmount()));
+    context.setVariable("remainingAmount", numberToReadable(dataProvider.getRemainingAmount()));
+    context.setVariable(
+        "paymentAmountAsWords", numberToWords(dataProvider.getTotalPaymentAmount()));
+    context.setVariable("paymentReason", dataProvider.getFeeComment());
+    context.setVariable("paymentType", paymentType(dataProvider.getPaymentType()));
+
+    return context;
+  }
+
+  private String paymentType(school.hei.haapi.endpoint.rest.model.Payment.TypeEnum typeEnum) {
+    return switch (typeEnum) {
+      case BANK_TRANSFER -> "Virement bancaire";
+      case CASH -> "En espèce";
+      case MOBILE_MONEY -> "Mobile Money";
+      case SCHOLARSHIP, FIX -> null;
```

can't you throw error if SCHOLARSHIP or FIX instead of null?